### PR TITLE
Refactoring of payment providers

### PR DESCRIPTION
**Pull request by Forster-Groux**
Some logic was duplicated in the payment providers :
- get connected account for a specific provider
- process transaction (stripe and privacy)
- privacy process transaction between script, webhook and cron
- error logs in webhooks

=> Add stripe transactions to reconcile transactions script

### CODE REVIEW

File: cron/hourly/reconcile-transactions.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+          await privacy.refreshCardDetails(card);
+        }
+      }
+
+      if (card.provider === 'STRIPE') {
+        const stripe = Stripe(host.slug === 'opencollective' ? config.stripe.secret : connectedAccount.token);
+
+        const { data: transactions } = await stripe.issuing.transactions.list({
+          card: card.id,
+          created: { gt: new Date(begin).getTime() },
```
Comment (znarf) [inline cron/hourly/reconcile-transactions.js:66]:
```suggestion
          created: { [Op.gt]: new Date(begin).getTime() },
```

File: server/paymentProviders/paypal/payouts.ts [hunk 2]
```diff
@@ -28,12 +30,7 @@ export const payExpensesBatch = async (expenses: typeof models.Expense[]): Promi
     throw new Error(`Could not find the host embursing the expense.`);
   }
 
-  const [connectedAccount] = await host.getConnectedAccounts({
-    where: { service: 'paypal', deletedAt: null },
-  });
-  if (!connectedAccount) {
-    throw new Error(`Host is not connected to PayPal Payouts.`);
-  }
+  const connectedAccount = await host.getAccountForPaymentProvider(providerName);
```
Comment (znarf) [inline server/paymentProviders/paypal/payouts.ts:33]: Nice

File: server/paymentProviders/utils.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
+import activities from '../constants/activities';
+import { types as CollectiveTypes } from '../constants/collectives';
+import ExpenseStatus from '../constants/expense_status';
+import ExpenseType from '../constants/expense_type';
+import { TransactionKind } from '../constants/transaction-kind';
+import { getFxRate } from '../lib/currency';
+import logger from '../lib/logger';
+import models from '../models';
+
+export const getVirtualCardForTransaction = async cardId => {
```
Comment (znarf) [inline server/paymentProviders/utils.js:10]: Typescript could be useful here. Maybe for a follow up.

**Review by znarf — COMMENTED** [review]
Looks good.

A small thing to correct (`gt`).

File: cron/hourly/reconcile-transactions.js (hunk 1 above)
Comment (Forster-Groux) [reply cron/hourly/reconcile-transactions.js:66]: C'est une requête stripe pas sequelize, cf : 
https://stripe.com/docs/api/issuing/transactions/list#list_issuing_transactions-created-gt

**Pull request merged**